agent/config/builder.go

```diff
@@ -1662,3 +1667,16 @@ func isUnixAddr(a net.Addr) bool {
 	_, ok := a.(*net.UnixAddr)
 	return ok
 }
+
+// isValidAltDomain returns true iff the given domain may be used
+// as an alternate domain name.
+func isValidAltDomain(domain, datacenter string) bool {
```

Comment:
Are there any tests to update for this? Or would a test in `dns_test.go` suffice?

Reply:
you could create `agent/config/builder_test.go` and test it separately.